controllers/slicegateway/slicegateway.go

```diff
@@ -1386,6 +1395,27 @@ func (r *SliceGwReconciler) ReconcileGatewayDeployments(ctx context.Context, sli
 				return ctrl.Result{}, err, true
 			}
 			return ctrl.Result{Requeue: true}, nil, true
+		} else {
+			// update logic for gateways
+			for i := range deployments.Items {
```

Comment:
The "if" conditional above goes over the list of deployments indirectly through the numGwInstances, so it is not appropriate to loop over the deployments afresh in the "else" block. Please move the image check outside the numGwInstances "for" loop. Create a new loop that goes over each deployment and update if there is an image mismatch.